src/interface/extension_interface.hpp

```diff
@@ -445,24 +477,79 @@ typename sb_handle_t::event_t _omatadd(
     element_t alpha, container_0_t a, index_t lda, element_t beta,
     container_1_t b, index_t ldb, container_2_t c, index_t ldc,
     const typename sb_handle_t::event_t& _dependencies) {
+  // bail out early if the leading dimensions are not correct
+  if (ldc < m || lda < (trans_a == 't' ? n : m) ||
+      ldb < (trans_b == 't' ? n : m)) {
+    typename sb_handle_t::event_t ret;
+    return ret;
+  }
```

Comment:
Probably it is better to make aware the user that there is an error with an exception, otherwise it is totally silent.

Reply:
will add a `throw std::invalid_argument ` here